src/maliput_malidrive/builder/builder_tools.h

```diff
@@ -149,29 +149,59 @@ std::vector<maliput::api::LaneEnd> SolveLaneEndsForInnerLaneSection(
     const maliput::api::RoadGeometry* rg, const maliput::api::LaneEnd& lane_end,
     const MalidriveXodrLaneProperties& xodr_lane_properties);
 
-/// Hold the travel direction of a lane obtained from parsing a userData XML node.
-/// An userData node example:
-/// @code{.xml}
-///  <userData>
-///      <vectorLane travelDir="undirected"/>
-///  </userData>
-/// @endcode
+/// Hold the travel direction of a lane from parsing an XML node.
 class LaneTravelDirection {
  public:
   MALIDRIVE_DEFAULT_COPY_AND_MOVE_AND_ASSIGN(LaneTravelDirection)
   LaneTravelDirection() = delete;
-
-  /// Constructs a LaneTravelDirection.
-  /// @param user_data Contains a UserData XMLNode that holds information
-  ///                  about the direction of the lane.
-  /// @throw maliput::common::assertion_error When the travel direction information is not compatible.
-  explicit LaneTravelDirection(const std::optional<std::string>& user_data);
-
   /// Possible directions obtained from the XODR.
   /// Note: "Undefined" refer to when travel direction information is omitted or node isn't well formed.
   enum class Direction { kUndirected = 0, kForward, kBackward, kBidirectional, kUndefined };
 
-  /// @returns The direction of the lane describend in the userData XML node.
+  explicit LaneTravelDirection(Direction direction) : travel_dir_(direction) {}
+
+  /// Constructs a LaneTravelDirection from userData.
+  /// @param user_data Contains a UserData XMLNode that holds information
+  ///                  about the direction of the lane.
+  /// An userData node example:
+  /// @code{.xml}
+  ///  <userData>
+  ///      <vectorLane travelDir="undirected"/>
+  ///  </userData>
+  /// @endcode
+  /// @returns A LaneTravelDirection.
+  /// @throw maliput::common::assertion_error When the travel direction information is not compatible.
+  static LaneTravelDirection CreateFromUserData(const std::optional<std::string>& user_data);
```

Comment:
nit: google style guide cpp.

Static methods goes first in the order of definitions

Reply:
Ha!, Thanks for pointing that out, I got confused with static constants and the fact this is a factory function.
As a factory function, it definitely has to go above constructor/assignements/destructor.